astropy/units/tests/test_quantity.py

```diff
@@ -458,13 +458,18 @@ def test_commutativity(self):
         assert self.q1 / u.s == u.Quantity(11.42, "m/s")
         assert u.s / self.q1 == u.Quantity(1 / 11.42, "s/m")
 
-    def test_power(self):
+    @pytest.mark.parametrize(
+        "exponent_type", [float, np.uint64, np.int32, np.float32, u.Quantity]
+    )
+    def test_power(self, exponent_type):
```

Comment:
This test does not actually test the regression, i.e. it doesn't fail on 6.0.1.

The difference is taking a power of a quantity vs. a power of a unit:

```
q = 5 * u.m
q = q**u.Quantity(3)
```
work in astropy 6.0.1,

```
q = 125 * u.m**u.Quantity(3)
```
does not
```

Reply:
Ah. My first test did correctly fail on 6.0.1
I will rehash this after lunch, thanks a lot for your attention to details !

Reply:
done !
note that maybe_simple_fraction is (I think) private API, so I focused my test on just public API by exercising various types of exponents in m**2, but the only case that actually failed on main (or v6.0.x) is still m**Quantity(2)

Reply:
Actually, we got the same regression with `u.Masked(2)`, so I included it in the test too